IPG-Automotive-UK/react-ui
# VE-252 Combine dark and light themes for mui v6

Closes [TD-252](https://sce.myjetbrains.com/youtrack/issue/VE-252/Combine-dark-and-light-theme-for-MUI-v6)

I also think this partially fixes an dark and light toggle sync issue with storybook we found [here](https://sce.myjetbrains.com/youtrack/issue/VE-248/Dark-mode-toggle-inconsistencies-in-Storybook) in `VirtoAppLayout` . This PR sync mode switch from storybook to `VirtoAppLayout` but not reverse.

## Changes

This PR continues migrating RUI into MUI v6, primarily focusing on enabling [cssVariables](https://mui.com/material-ui/customization/css-theme-variables/configuration/) and [cssColorSchemes](https://mui.com/material-ui/customization/palette/#color-schemes) to keep up with industry standards and for an improved developer experience.

**ThemeProvider**
- Combined `defaultTheme` , `darkTheme` and `lightTheme` to `mainTheme` with `colorSchemes` API
- Changed `--mui` default variable prefix to `--ipg` to label the theme brand. With `cssVariables` enabled anyone can easily check where the tokens come from now before it just showed the color code used.

See an example of how the background-color token displays now in dev tools
Before
![Screenshot 2024-12-10 155711](https://github.com/user-attachments/assets/e6bd82f9-a9ee-476c-aabf-d6c8f884970b)
After
![Screenshot 2024-12-10 155229](https://github.com/user-attachments/assets/db60cb2d-22bc-4b76-ab8f-7e0a0ef856ff)

- Update the `Theme` interface to support `colorSchemes`
- Rendering `ThemProvider` `children` inside a new `ChildWrapper` component to make use of `useColorScheme`
- Remove `ThemeContext.tsx`  which used to support use of `useTheme` hook throughout the app to get and set the current theme, instead of using the new `useColorScheme` coming with MUI v6. This includes fixes in tests as well.
- Replaced few custom colors with `var(--ipg-palette...)` from MUI (providing the intented color)

**ToggleColorMode**
- Using the `useColorScheme` to get and set the theme
- Updated tests

**VirtoAppHeader**
- Stories in `VirtoAppHeader.stories.tsx` updated as we no longer needed to pass `mode`
- `onColourModeChange` prop was removed and tests were removed as we no longer pass the props `mode`

**Some other file changes**
- Fixed a failing test in `ConfirmProvider.test.tsx`
- Updated `VirtoLogo.jsx` prop to stop throwing warnings by passing sx with new theme conversion type as function
- Background color of content shown part in `VirtoAppLayout` now used `var(--ipg-palette-background-paper)` to enable correct background.

## UI/UX

N/A

## Testing notes

- Make sure all components still work the same as before. 
- Make sure all color codes now work the same as before.

## Author checklist

Before I request a review:
